### DISCUSSION TITLE
Is it possible to "download" the archive@blockNumber, and run it locally?

### DISCUSSION THREAD

**Discussion by emilyJLin95** [ConsenSys-archive/ganache#3447] [Q&A] [answered]
Reposting from the Truffle Discord:
I want to "fork" mainnet at a specific blockNumber. I tried Ganache with Infura, but it rate-limited my requests.

Is it possible to "download" the archive@blockNumber, and run it locally?

**Comment by davidmurdoch** [comment, marked as answer]
Ganache's built-in forking has a higher rate limit than the free versions. It is usually very difficult to hit our limit under normal circumstances.

Also, Ganache does download the state it needs at the specified `blockNumber` and will reused that state later. However, if the `blockNumber` changes between runs it has to re-fetch the data all over again.

**Reply by MicaiahReid** [reply]
To add onto this, Ganache's built-in forking can be used by running `ganache --fork` (to fork mainnet) or `ganache --fork ropsten` (ropsten can be replaced with another testnet).
